iree/compiler/Dialect/LinalgPlus/IR/LinalgPlusOps.td

```diff
@@ -0,0 +1,63 @@
+// Copyright 2021 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#ifndef IREE_DIALECT_LINALGPLUS_OPS
+#define IREE_DIALECT_LINALGPLUS_OPS
+
+include "iree/compiler/Dialect/LinalgPlus/IR/LinalgPlusBase.td"
+include "mlir/Interfaces/SideEffectInterfaces.td"
+include "mlir/Interfaces/ControlFlowInterfaces.td"
+
+//===----------------------------------------------------------------------===//
+// Base class.
+//===----------------------------------------------------------------------===//
+
+class LinalgPlus_Op<string mnemonic, list<OpTrait> traits = []> :
+    Op<LinalgPlus_Dialect, mnemonic, traits> {
+  let verifier = [{ return verify$cppClass(*this); }];
+}
+
+//===----------------------------------------------------------------------===//
+// Non-structured ops
+//===----------------------------------------------------------------------===//
+
+def LinalgPlus_SortOp : LinalgPlus_Op<"sort", [RecursiveSideEffects,
```

Comment:
Do we need `RecursiveSideEffects`. I think whatever attributes the `linalg.generic` op upstream has should be enough. I only see

```
   DeclareOpInterfaceMethods<MemoryEffectsOpInterface>,
   SingleBlockImplicitTerminator<"YieldOp">
```

Reply:
Yes, I reference to Linalg ops and have the same config. In this case, we need to implement `getEffects` and we can have the same form as Linalg structured ops.